ikarus/finiteelements/mechanics/materials/hyperelastic/deviatoric/gent.hh

```diff
@@ -0,0 +1,188 @@
+ * invariant \f$ W_1 \f$, see \ref DeviatoricInvariants.
+ *
+ * \tparam ST_ The underlying scalar type.
+ */
+template <typename ST_>
+struct GentT
+{
+  static constexpr int dim = 3;
+  using ScalarType         = ST_;
+
+  template <typename ST = ScalarType>
+  using PrincipalStretches = Eigen::Vector<ST, 3>;
+
+  template <typename ST = ScalarType>
+  using Invariants = PrincipalStretches<ST>;
+
+  template <typename ST = ScalarType>
+  using FirstDerivative = Eigen::Vector<ST, dim>;
+  template <typename ST = ScalarType>
+  using SecondDerivative = Eigen::Matrix<ST, dim, dim>;
+
+  using MaterialParameters = GentMatParameters;
+
+  static constexpr auto stretchTag = PrincipalStretchTags::deviatoric;
+
+  [[nodiscard]] constexpr static std::string name() noexcept { return "Gent"; }
+
+  /**
+   * \brief Constructor for GentT
+   *
+   * \param C material constant
+   * \param lambdaM maximum stretch at which the polymer chain locks
+   */
+  explicit GentT(const MaterialParameters& matPar)
+      : matPar_{matPar} {}
+
+  /**
+   * \brief Returns the material parameters stored in the material
+   */
+  MaterialParameters materialParametersImpl() const { return matPar_; }
+
+  /**
+   * \brief Computes the stored energy in the Gent material model.
+   * \details Using only the first five terms of the inverse Langevin function.
+   *
+   * \param lambda principal stretches
+   * \tparam ST the scalartype of the principal stretches
+   * \return ScalarType
+   */
+  template <typename ST = ScalarType>
+  ST storedEnergyImpl(const PrincipalStretches<ST>& lambda) const {
+    const Invariants<ST>& invariants = Impl::invariants(lambda);
+    const auto& devInvariants        = DeviatoricInvariants<PrincipalStretches<ST>>(lambda);
+    const auto W1                    = devInvariants.value()[0];
+    checkJm(W1);
+    return -(matPar_.mu / 2.0) * matPar_.Jm * log(1.0 - ((W1 - 3.0) / matPar_.Jm));
+  }
+
+  /**
+   * \brief Computes the first derivative of the stored energy function w.r.t. the total principal stretches.
+   *
+   * \param lambda principal stretches
+   * \tparam ST the scalartype of the principal stretches
+   * \return ScalarType
+   */
+  template <typename ST = ScalarType>
+  FirstDerivative<ST> firstDerivativeImpl(const PrincipalStretches<ST>& lambda) const {
+    const Invariants<ST>& invariants = Impl::invariants(lambda);
+    auto dWdLambda                   = FirstDerivative<ST>::Zero().eval();
+
+    const auto& devInvariants = DeviatoricInvariants<PrincipalStretches<ST>>(lambda);
+    const auto W1             = devInvariants.value()[0];
+    const auto& dW1dLambda    = devInvariants.firstDerivative().first;
+    const auto mu             = matPar_.mu;
+    const auto Jm             = matPar_.Jm;
+    checkJm(W1);
+
+    for (const auto k : dimensionRange())
+      dWdLambda[k] += (mu * dW1dLambda[k] * Jm) / (2.0 * (Jm - W1) + 6.0);
```

Comment:
This does not need to be a += since it only adds to zero and also this can simplified without indices.